File: golang-devildyw/middle/L102二叉树的层序遍历.go
package main

func main() {}

func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return [][]int{}
	}

	result := [][]int{}
	queue := []*TreeNode{root}

	for len(queue) > 0 {
		length := len(queue)
		temp := []int{}
		for length > 0 {
			first := queue[0]
			queue = queue[1:]
			temp = append(temp, first.Val)
			if first.Left != nil {
				queue = append(queue, first.Left)
			}
			if first.Right != nil {
				queue = append(queue, first.Right)
			}
			length--
		}
		result = append(result, temp)
	}
	return result
}

File: java-devildyw/src/main/java/top/dyw/leetcode/Hot100/L102二叉树的层序遍历.java
package top.dyw.leetcode.Hot100;
import java.util.*;
public class L102二叉树的层序遍历 {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<>();
        if (root == null) {
            return result;
        }

        Deque<TreeNode> queue = new ArrayDeque<>();
        queue.offer(root);

        while (!queue.isEmpty()) {
            int size = queue.size();
            List<Integer> temp = new ArrayList<>();
            while (size-->0) {
                TreeNode node = queue.poll();
                temp.add(node.val);
                if (node.left!=null) {
                    queue.offer(node.left);
                }
                if (node.right!=null) {
                    queue.offer(node.right);
                }
            }
            result.add(temp);
        }

        return result;
    }
}
